knowledge-base/qrcode-barcode-chart-export-to-image.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Export QRCode, Barcode, or Chart to Image
+description: Learn how to convert and export a Telerik Blazor QRCode, Barcode, or Chart to an image and send it to the .NET runtime.
+type: how-to
+page_title: How to Export Telerik QRCode, Barcode, or Chart to Image
+slug: qrcode-barcode-chart-kb-export-to-image
+tags: telerik, blazor, qrcode, barcode, chart
+ticketid: 1572189, 1588186, 1667798
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>
+                QRCode for Blazor, <br />
+                Barcode for Blazor, <br />
+                Chart for Blazor
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This KB answers the following questions:
+
+* How to export a generated QRCode to a PNG or JPG image for saving on the server or download?
+* How to create an image from the Telerik QRCode for Blazor?
+* How to convert a Barcode, Chart, or QRCode from SVG to an image?
+* How to save a Barcode or QRCode as an image?
+
+## Solution
+
+Use `JSInterop` and JavaScript APIs to convert the Telerik component to a Base64 string and send it to the server.
+
+When using the `Canvas` `RenderingMode` of the Telerik component (Barcode, Chart, QRCode), use the [`toDataURL`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) method of the `canvas` HTML element to obtain a Base64 data URI for the image.
+
+When using the `SVG` rendering mode:
+
+1. Use an [`XMLSerializer`](https://developer.mozilla.org/en-US/docs/Web/API/XMLSerializer) to create an image [`Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob).
+1. Create a `canvas` element and [get its 2D context](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext).
+1. [Draw the image](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage) into the `canvas` element.
+1. Use the [`toDataURL`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) method of the `canvas` HTML element to obtain a Base64 data URI for the image.
+
+> When using a Blazor app with **Server** render mode, make sure to [increase the SignalR max message size]({%slug common-kb-increase-signalr-max-message-size%}), otherwise the Base64 data URI may not reach the .NET runtime.
+
+>caption Export QRCode, BarCode or Chart to Image
+
+````CSHTML
```

Comment:
Let's start using RAZOR instead of CSHTML. Due to the docs modernization initiative, we already started replacing CSHTML with RAZOR in all docs files.